Classes/System/Solr/SolrConnection.php

```diff
@@ -90,7 +98,14 @@ class SolrConnection
     protected $logger = null;
 
     /**
-     * @var array
+     * Adapter class used for the communication
+     *
+     * @var string
+     */
+    protected $adapterClass = Curl::class;
```

Comment:
We should fix and switch back to the TYPO3 Core functionality by using Guzzle
https://docs.typo3.org/m/typo3/reference-coreapi/master/en-us/ApiOverview/Http/Index.html
To be able to configure the outgoing requests behind strict networks.

Reply:
Currently Guzzle is not part of Solarium, so this need an own implementation.

Reply:
The class Psr18Adapter itself requires a newer Guzzle version that TYPO3 not support in the current LTS version.